src/mrest.cpp

```diff
@@ -79,4 +84,55 @@ int MRest::ResetHorizontalAlignment(FunctorParams *functorParams)
     return FUNCTOR_CONTINUE;
 }
 
+int MRest::GetOptimalLayerLocation(Staff* staff, Layer* layer, int defaultLocation)
+{
+    Staff *parentStaff = vrv_cast<Staff *>(GetFirstAncestor(STAFF));
+    if (!layer) return defaultLocation;
+    // handle rest positioning for 2 layers. 3 layers and more are much more complex to solve
+    if (parentStaff->GetChildCount(LAYER) != 2) return defaultLocation;
+
+    ListOfObjects layers;
+    ClassIdComparison matchType(LAYER);
+    parentStaff->FindAllDescendantByComparison(&layers, &matchType);
+    const bool isTopLayer = (vrv_cast<Layer *>(*layers.begin())->GetN() == layer->GetN());
+
+    auto otherLayerIter = isTopLayer ? std::prev(layers.end()) : layers.begin();
+    auto collidingElementsList = vrv_cast<Layer *>(*otherLayerIter)->GetLayerElementsForTimeSpanOf(this);
```

Comment:
Please do not use `auto` in such cases because it makes the code harder to read.